### PR TITLE
DDS-1190 code coverage

### PR DESCRIPTION
**Pull request by dmann**
- Use SimpleCov to calculate code coverage
- Adding `coverage` workflow to CI to merge parallel coverage results and store in job's artifacts

### CODE REVIEW

File: .circleci/report_coverage.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env ruby
```
Comment (dmlond) [inline .circleci/report_coverage.rb:1]: I dont know if we need to be consistent, but in other projects where I have created executables for use in CI, I have used `bin/ci` to store the executables.

**Review by dmlond — CHANGES_REQUESTED** [review]

File: .circleci/report_coverage.rb (hunk 1 above)
Comment (dmann) [reply .circleci/report_coverage.rb:1]: Currently the CI executables for this project are in the `.circleci` directory. I'm fine with refactoring the directory structure for the project to follow a more generic standard, but I think that's out of the scope of this PR. Maybe make a card in Jira and we can address this issue, as well as the directory structure as a whole.

Comment (dmlond) [reply .circleci/report_coverage.rb:1]: ok. that's fine. I just wanted to bring it up before it was too late

**Review by dmlond — APPROVED** [review]
💯

**Pull request merged**